growthbook/growthbook
# Clean up orphaned feature rules on env deletion

### Features and Changes

Currently when an environment is deleted, any feature rules that were created for that environment are left intact. This adds a quick cleanup to delete them (to prepare for upcoming changes which will allow the creation of many more environments)

### Testing

Checked in a mongo shell to confirm deletion behavior:

1. Create a new environment `staging`
2. Create a rule on a feature in the `staging` environment
3. Run the following in a mongo shell
```
const environmentKey = `environmentSettings.staging`;
const query = {[environmentKey]: {$exists: true}}
db.features.find(query)
```
You should see the feature whose rule you just created.
4. Delete the environment in the UI or by API request, leaving `removeAssociatedFeatureRules` empty or false
5. Re-run `db.features.find(query)`; the feature should still be there
6. Re-create the environment `staging` in the UI
7. Run the delete again, this time setting `removeAssociatedFeatureRules` to true
8. Re-run `db.features.find(query)` and observe that there are no results

### Screenshots
![image](https://github.com/user-attachments/assets/91e81fc3-3042-45ce-9cb9-d86ecc30876b)
